app/jobs/get_suggestions_job.rb

```diff
@@ -0,0 +1,93 @@
+class GetSuggestionsJob < ApplicationJob
+  queue_as :default
+
+  def perform(job_status)
+    @job_id = job_status.job_id
+
+    send_spotify_request
+
+    if get_spotify_tracks
+      job_status.update(job_attributes_success)
+    else
+      job_status.update(job_attributes_failure)
```

Comment:
This calls `#update`, and then `#save!` again. How about merging `done: true` into the `#job_attributes_...` methods?